rotkehlchen/globaldb/handler.py

```diff
@@ -1011,6 +1011,123 @@ def add_historical_prices(entries: List['HistoricalPrice']) -> None:
 
         connection.commit()
 
+    @staticmethod
+    def add_single_historical_price(entry: HistoricalPrice) -> None:
+        """Adds the given historical price entries in the DB
+
+        If any addition causes a DB error it's skipped and an error is logged
+        """
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        try:
+            serialized = entry.serialize_for_db()
+            cursor.execute(
+                'SELECT COUNT(*) FROM price_history WHERE from_asset=? AND '
+                'to_asset=? AND timestamp=?',
+                (serialized[0], serialized[1], serialized[3]),
+            )
+            if cursor.fetchone()[0] is None:
+                return
+            cursor.execute(
+                """INSERT OR IGNORE INTO price_history(
+                from_asset, to_asset, source_type, timestamp, price
+                ) VALUES (?, ?, ?, ?, ?)
+                """,
+                serialized,
+            )
```

Comment:
Don't do that. Simply try to add it. If there is an error then you know it already exists in there or if the error is a foreign key constraint that the from/to asset do not exist in the DB (this last one should not happen if we don't have a bug somewhere else).